lib/src/wip/autocomplete/popup.dart

```diff
@@ -27,18 +26,24 @@ class Popup extends StatefulWidget {
   }) : super(key: key);
 
   @override
-  _PopupState createState() => _PopupState();
+  PopupState createState() => PopupState();
 }
 
-class _PopupState extends State<Popup> {
-  late double width;
-  late double height;
+class PopupState extends State<Popup> {
+  static const double _maxWidth = Sizes.autocompletePopupMaxWidth;
+  static const double _maxHeight = Sizes.autocompletePopupMaxHeight;
+  bool _verticalOverflow = false;
+  bool _horizontalOverflow = false;
 
   @override
   void initState() {
     widget.controller.addListener(rebuild);
-    this.width = 300;
-    this.height = 100;
+    _verticalOverflow =
```

Comment:
These should not be stored because they can change after `initState()`. Make them local variables in `build()`.